src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -36,117 +32,137 @@
+import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @ExtendWith({MockitoExtension.class, GGExtension.class})
 public class CustomCaConfigurationTest {
     @Mock
-    private SessionManager sessionManagerMock;
-
+    SecurityService securityServiceMock;
     @Mock
-    private GroupManager groupManagerMock;
-
-    @Mock private CertificateRegistry certificateRegistryMock;
-
+    private GreengrassServiceClientFactory clientFactory;
     @Mock
-    ConnectivityInformation mockConnectivityInformation;
-
+    private GroupManager groupManager;
     @Mock
-    CertificateExpiryMonitor mockCertExpiryMonitor;
-
+    CertificateExpiryMonitor certExpiryMonitor;
     @Mock
-    CISShadowMonitor mockShadowMonitor;
-
-    @Mock
-    GreengrassServiceClientFactory clientFactoryMock;
-    @Mock
-    SecurityService securityServiceMock;
+    private GreengrassV2DataClient client;
     @TempDir
-    Path tmpPath;
-
-    private CertificateManager certificateManager;
-    private CertificateStore certificateStore;
+    Path rootDir;
+    private Kernel kernel;
 
 
     @BeforeEach
-    void beforeEach() {
-        DomainEvents events = new DomainEvents();
-        certificateStore = new CertificateStore(tmpPath, events);
+    void setup(ExtensionContext context) throws DeviceConfigurationException {
+        ignoreExceptionOfType(context, SpoolerStoreException.class);
+
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+        kernel.getContext().put(GroupManager.class, groupManager);
+        kernel.getContext().put(SecurityService.class, securityServiceMock);
+        kernel.getContext().put(CertificateExpiryMonitor.class, certExpiryMonitor);
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+
+        lenient().when(clientFactory.fetchGreengrassV2DataClient()).thenReturn(client);
+    }
 
-        certificateManager = new CertificateManager(
-                certificateStore, mockConnectivityInformation,
-                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), clientFactoryMock, securityServiceMock);
+    @AfterEach
+    void cleanup() {
+        kernel.shutdown();
+    }
 
+    // TODO: Consolidate this test helpers with ClientDevicesAuthServiceTest
+    private void startNucleus() throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        Path resourceDirectory = Paths.get(
```

Comment:
You can just use `<class>.getResourceURI`

Reply:
I tried that but it didn't work, got a NullPointerException, This test is inside the integrationtests folder. Is there something I am missing?. Basically `getClass().getResource(configFileName)` and didn't work

Reply:
Sounds like the yaml file is in the wrong place then. `integrationtests` should have its own `resources` directory.

Might also need this: https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/c950a691d92f12204d447b021891253d756fd5ec/pom.xml#L328-L341